src/main/java/it/pagopa/transactions/utils/JwtTokenUtils.java

```diff
@@ -0,0 +1,61 @@
+package it.pagopa.transactions.utils;
+
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.crypto.SecretKey;
+import java.security.SecureRandom;
+import java.util.Calendar;
+import java.util.Date;
+
+@Component
+@Slf4j
+public class JwtTokenUtils {
+
+    private final SecretKey jwtSecretKey;
+    private final SecureRandom secureRandom = new SecureRandom();
+
+    public static final String TRANSACTION_ID_CLAIM = "transactionId";
+
+    private final int tokenValidityTimeMillis;
+
+    public JwtTokenUtils(
+            @Autowired SecretKey jwtSecretKey,
+            @Value("${jwt.validityMillis}") int tokenValidityMillis
+    ) {
+        this.jwtSecretKey = jwtSecretKey;
+        this.tokenValidityTimeMillis = tokenValidityMillis;
+    }
+
+    public String generateToken(String transactionId) {
+        try {
+            Calendar calendar = Calendar.getInstance();
+            Date issuedAtDate = calendar.getTime();
+            calendar.add(Calendar.MILLISECOND, tokenValidityTimeMillis);
+            Date expiryDate = calendar.getTime();
+            return Jwts.builder()
+                    .claim(TRANSACTION_ID_CLAIM, transactionId)// transactionId (custom claim)
+                    .setId(generateNonce())// jti
+                    .setIssuedAt(issuedAtDate)// iat
+                    .setExpiration(expiryDate)// exp
+                    .signWith(jwtSecretKey)
+                    .compact();
+        } catch (JwtException e) {
+            log.error("Error generating JWT token", e);
+            return null;
+        }
+
+    }
+
+    private String generateNonce() {
+        StringBuilder nonce = new StringBuilder();
+        for (int curr = 0; curr < 15; curr++) {
+            nonce.append(secureRandom.nextInt(10));
+        }
+        return nonce.toString();
+    }
```

Comment:
Just a quick comment (that maybe should be included in code to validate our assumptions in the future 🙂)

There are $10^{15}$ possibilities for a nonce
Each nonce is valid for 30 minutes (a session duration, currently the same as the duration of a payment token)

Assuming 100 HTTP req/s for an attacker (for a single IP address it's an infrastructure constraint we put, in a more advanced threat model we could assume a distributed attack), we have
100 rps * (30 * 60) = 180 k guesses

$p(\text{1 right guess}) = \frac{1}{10^{15}}$
$p(\text{right guess before expiry}) = \frac{1.8 * 10^5}{10^{15}} = 1.8 * 10^{-10} \approx 2^{-29}$

We'll have to discuss with security whether this value is enough, but given how short these tokens' durations are as a first approximation I'd say it's ok

Reply:
Generated tokens are signed with HMAC key retrieved from KV that must be at least 256-bit long as stated by `Keys.hmacShaKeyFor` that will throw `WeakKeyException` for shorter keys. 
As for now, if the secret configured key is an invalid base64 key or shorter than the length expressed above then the module startup fails.

Considering that the tokens are signed with HMAC and that into the token body is present also the transactionId claim that is a random UUID generated for each transaction I would say that is not a problem.

Beside that, since [jti token claim](https://www.rfc-editor.org/rfc/rfc7519#section-4.1.7) is intended to be an unique id for the token (mainly used for token revocations) I changed the jti claim to be an UUID so that it can be identified univocally.